src/components/Personalization/createOnResponseHandler.js

```diff
@@ -16,7 +16,7 @@ const DECISIONS_HANDLE = "personalization:decisions";
 export default ({ extractDecisions, executeDecisions, showContainers }) => {
   return ({ renderDecisions, response }) => {
     const unprocessedDecisions = response.getPayloadsByType(DECISIONS_HANDLE);
-    if (!isNonEmptyArray(unprocessedDecisions)) {
+    if (renderDecisions && !isNonEmptyArray(unprocessedDecisions)) {
```

Comment:
This isn't a big deal, but consider switching out `!isNonEmptyArray(unprocessedDecisions)` for `unprocessedDecisions.length`. We know `unprocessedDecisions` will always be an array and `!isNonEmptyArray` involves more mental load than necessary (it's a double negative and can also lead the reader to wonder whether we're checking if `unprocessedDecisions` has items or whether `unprocessedDecisions` is not an array or both).

Reply:
I noticed you have another condition with `renderDecisions` down below. We may want to consolidate all this. If I understand the logic correctly, maybe something like this?

```
let unprocessedDecisions = response.getPayloadsByType(DECISIONS_HANDLE);

if (renderDecisions) {
  let renderableDecisions;
  ([renderableDecisions, unprocessedDecisions] = extractDecisions(unprocessedDecisions));
  if (renderableDecisions.length) { // Not sure if you need this. If you need this check it all, it would probably be best to move it to inside executeDecisions
    executeDecisions(renderableDecisions);
  }
  showContainers();
}

return { decisions: unprocessedDecisions };
```

Reply:
Have these been thought through:

Do we always want to add `decisions` to the object that customers receive in response to executing a `sendEvent` command? Are we doing so currently? Is the value always an array? Do we have tests that ensure that this is the case?

Reply:
we return in the `sendEvent` promise the decisions array that contains decisions that can't be rendered by us. `decisions` is always an array.

Reply:
It might be good to have a functional test for this. Specifically, if no decisions are returned that can't be auto-rendered, ensure that the `result` object that comes through the promise still has a `decisions` array. It would also be helpful to update [the personalization documentation](https://docs.adobe.com/content/help/en/experience-platform/edge/fundamentals/rendering-personalization-content.html) to show that `decisions` will always be provided and will always be an array. Our documentation currently suggests that `result.decisions` is only sometimes provided, because we have this example that checks if `result.decisions` exists:

```
alloy("sendEvent",{
    xdm:{...},
    decisionScopes:['demo-1', 'demo-2']
  }).then(function(result){
    if (result.decisions){
      // Do something with the decisions.
    }
  })
```

Reply:
`decisions` will always be provided if personalization on the call.